metadata-ingestion/src/datahub/ingestion/source/ge_profiling_config.py

```diff
@@ -145,10 +145,26 @@ class GEProfilingConfig(ConfigModel):
     # Hidden option - used for debugging purposes.
     catch_exceptions: bool = Field(default=True, description="")
 
-    partition_profiling_enabled: bool = Field(default=True, description="")
+    partition_profiling_enabled: bool = Field(
+        default=True,
+        description="Whether to profile partitioned tables. Only BigQuery supports this. "
+        "If enabled, latest partition data is used for profiling.",
+    )
     partition_datetime: Optional[datetime.datetime] = Field(
         default=None,
-        description="For partitioned datasets profile only the partition which matches the datetime or profile the latest one if not set. Only Bigquery supports this.",
+        description="If specified, profile only the partition which matches this datetime. "
+        "If not specified, profile the latest partition. Only Bigquery supports this.",
+    )
+    use_sampling: bool = Field(
+        default=True,
+        description="Whether to profile column level stats on sample of table. Only BigQuery supports this. "
+        "If enabled, profiling is done on rows sampled from table. Sampling is not done for smaller tables. ",
+    )
+
+    sample_size: int = Field(
+        default=1000,
```

Comment:
How'd you decide on this default? Do you know approx how long each 1000, 10000, 100000, etc. rows will take? I wonder if 10,000 is a better default, for higher fidelity with a small tradeoff in speed.

Reply:
I don't have performance numbers on different sizes. I'm open to changing this default. We can maybe come back to this after we have some numbers on speed and some observations on fidelity based on real world data.